packages/cubejs-redshift-driver/src/RedshiftDriver.ts

```diff
@@ -122,24 +150,27 @@ export class RedshiftDriver extends PostgresDriver<RedshiftDriverConfiguration>
         }
       });
 
+      const baseQuery = `UNLOAD ('SELECT * FROM ${table}') TO '${bucketType}://${bucketName}/${exportPathName}/'`;
+      const credentialQuery = unloadArn ?
+        `iam_role '${unloadArn}'` : `CREDENTIALS 'aws_access_key_id=${keyId};aws_secret_access_key=${secretKey}'`;
+      const unloadQuery = `${baseQuery} ${credentialQuery} ${optionsPart}`;
+
       // Unable to extract number of extracted rows, because it's done in protocol notice
       await conn.query({
-        text: (
-          `UNLOAD ('SELECT * FROM ${table}') TO '${bucketType}://${bucketName}/${exportPathName}/' ` +
-          `CREDENTIALS 'aws_access_key_id=${keyId};aws_secret_access_key=${secretKey}' ${optionsPart}`
-        ),
+        text: unloadQuery,
       });
+
       if (unloadTotalRows === 0) {
         return {
           csvFile: [],
         };
       }
 
       const client = new S3({
-        credentials: {
+        credentials: (keyId && secretKey) ? {
```

Comment:
Another note,  this behavior can be tested by launching an EC2 instance with an attached `iam_role` that has access to a single S3 bucket.

You can install the AWS CLI, and without configuration (setting keys/secrets) you should only be able to access that assigned bucket.

Reply:
Cool! Thanks for the clarification. Can you please add the comment(s) content as documentation for `RedshiftDriverExportArnAWS.unloadArn`. For example, but feel free to provide a better blurb :) 

```
 interface RedshiftDriverExportArnAWS extends RedshiftDriverExportRequiredAWS {
   // ARN used to access S3 data from e.g. EC2 instances, instead of explicit key/secret credentials.
   unloadArn?: string,
 }
```